openedx/features/course_experience/tests/views/test_course_home.py

```diff
@@ -321,6 +322,11 @@ def test_home_page(
             if expected_enroll_message:
                 self.assertContains(response, 'You must be enrolled in the course to see course content.')
 
+        if enable_unenrolled_access and course_visibility == COURSE_VISIBILITY_PUBLIC:
+            if not is_enrolled_or_staff and not is_anonymous and self.course.invitation_only:
```

Comment:
Nit: You can use the `user_type` sent in the input. So this check can be changed to:
```
if user_type == CourseUserType.UNENROLLED and self.course.invitation_only:
```

Reply:
I will update.